### PR TITLE
Ticket1274 time channel settings in dae tab (GUI)

### PR DESCRIPTION
**Pull request by ThomasLohnert**
https://github.com/ISISComputingGroup/IBEX/issues/1274

Can now select between time channel settings from file or from parameters specified in GUI.

### CODE REVIEW

**Comment by ThomasLohnert** [comment]
Related:
https://github.com/ISISComputingGroup/EPICS-ioc/pull/47
https://github.com/ISISComputingGroup/EPICS-isisdae/pull/6

File: base/uk.ac.stfc.isis.ibex.dae/src/uk/ac/stfc/isis/ibex/dae/experimentsetup/ExperimentSetup.java [hunk 1]
```diff
@@ -81,6 +83,10 @@ public DaeSettings daeSettings() {
 		return periodFiles;
 	}
 	
+    public UpdatedValue<Collection<String>> timeChannelFiles() {
```
Comment (John-Holt-Tessella) [inline base/uk.ac.stfc.isis.ibex.dae/src/uk/ac/stfc/isis/ibex/dae/experimentsetup/ExperimentSetup.java]: Could you add some java doc.

File: base/uk.ac.stfc.isis.ibex.ui.dae/src/uk/ac/stfc/isis/ibex/ui/dae/experimentsetup/timechannels/TimeChannelsPanel.java [hunk 2]
```diff
@@ -121,4 +149,13 @@ private void clearExistingTimeRegimeViews() {
 		}
 		timeRegimeViews.clear();
 	}
+
```
Comment (John-Holt-Tessella) [inline base/uk.ac.stfc.isis.ibex.ui.dae/src/uk/ac/stfc/isis/ibex/ui/dae/experimentsetup/timechannels/TimeChannelsPanel.java]: I think these lines are from a previous incarnation of the change can you delete these.

File: base/uk.ac.stfc.isis.ibex.ui.dae/src/uk/ac/stfc/isis/ibex/ui/dae/experimentsetup/timechannels/TimeChannelsViewModel.java [hunk 3]
```diff
@@ -80,4 +122,16 @@ public void setTimeUnit(int index) {
...
+
+    /**
+     * Used to convert an UpdatedValue with type string collection into a string
+     * array.
+     * 
+     * @param updated the UpdatedValue object to be converted
+     * @return the String array
+     */
+    private String[] valueOrEmpty(UpdatedValue<Collection<String>> updated) {
+        Collection<String> value = updated.getValue();
```
Comment (John-Holt-Tessella) [inline base/uk.ac.stfc.isis.ibex.ui.dae/src/uk/ac/stfc/isis/ibex/ui/dae/experimentsetup/timechannels/TimeChannelsViewModel.java]: excellent catch of case where there are no files

**Pull request merged**